cmd/juju/application/deploy.go

```diff
@@ -213,6 +231,14 @@ func (a *deployAPIAdapter) GetAnnotations(tags []string) ([]apiparams.Annotation
 	return a.annotationsClient.Get(tags)
 }
 
+type charmstoreClientShim struct {
+	*csclient.Client
+}
+
+func (c *charmstoreClientShim) WithChannel(channel csparams.Channel) charmstoreForDeploy {
```

Comment:
Unless I'm missing something I'm not sure what we need the shim for if this is the only method on it

Reply:
The code won't compile without it. csclient.Client.WithChannel() returns a *csclient.Client, rather than a charmstoreForDeploy.. happy to take suggestions.